File: backend/routes/mobile.js
const express = require('express');
const router = express.Router();
const { searchPNBApps, scanMobileApp } = require('../services/mobile-scanner');

// Search for PNB Apps
router.get('/search', async (req, res) => {
  try {
    const apps = await searchPNBApps();
    res.json({ success: true, apps });
  } catch (error) {
    res.status(500).json({ success: false, error: error.message });
  }
});

// Scan a specific App
router.post('/scan', async (req, res) => {
  const { appId, platform } = req.body;
  try {
    const results = await scanMobileApp(appId, platform);
    res.json({ success: true, results });
  } catch (error) {
    res.status(500).json({ success: false, error: error.message });
  }
});

module.exports = router;

File: backend/routers/mobile.py
"""
Mobile scanner router.
"""

from fastapi import APIRouter
from pydantic import BaseModel
from typing import Optional

from services.mobile_scanner import search_pnb_apps, scan_mobile_app

router = APIRouter()


class MobileScanRequest(BaseModel):
    appId: str
    platform: str = "Android"


@router.get("/search")
def search():
    apps = search_pnb_apps()
    return {"success": True, "apps": apps}


@router.post("/scan")
def scan(body: MobileScanRequest):
    results = scan_mobile_app(body.appId, body.platform)
    return {"success": True, "results": results}
